core/container.go

```diff
@@ -145,6 +145,27 @@ func (mnt ContainerMount) SourceState() (llb.State, error) {
 	return defToState(mnt.Source)
 }
 
+type ContainerMounts []ContainerMount
+
+func (mnts ContainerMounts) With(newMnt ContainerMount) ContainerMounts {
+	mntsCp := make(ContainerMounts, len(mnts))
+	copy(mntsCp, mnts)
+
+	var replaced bool
+	for i, mnt := range mntsCp {
+		if mnt.Target == newMnt.Target {
+			mntsCp[i] = newMnt
```

Comment:
Might be better to remove from the found index and append to the end instead to match stacking semantics, i.e. so later mounts to subpaths of the target don't mount "over" it. Not sure if that's what'll actually happen though.

Reply:
Oh right, so it doesn't really matter where we insert.

Added a test that checks that when adding mount B above, mount A is removed and we can still add mount C after. I think this is more correct than the current behavior which allows it to stick around. But we're really splitting hairs, I'd be equally happy to remove this and leave the behavior undefined until someone has a use case. :shrug: